### PR TITLE
chore(deps): update bluebird

### LINKED ISSUE #1674: Upgrade to bluebird version 3
Currently this repo is stranded on version 2 of bluebird because it breaks with the version 3 changes to `.done` ([release notes](http://bluebirdjs.com/docs/new-in-bluebird-3.html)).

In order to upgrade to version 3 we either need to:

* Ensure that no unhandled rejections ever make it to `done`, or;
* Replace calls to `.done` with `.then`.

The second option seems safer and easier to implement.

### PR DESCRIPTION
**Pull request by philbooth**
Fixes #1674.

There are three breaking changes in particular that affect us:

* Unhandled rejections that reach `done` kill the process. To that end, I've replaced calls to `done` with `then`.

* The second argument to `promisify` is an options object rather than the invocation context. The invocation context is specified via `options.context`.

* Promisification methods ignore multiple callback arguments and resolve the first argument only. There is a new `multiArgs` option to control this.

You can read more about the wider bluebird changes [here](http://bluebirdjs.com/docs/new-in-bluebird-3.html).

@mozilla/fxa-devs r?

### CODE REVIEW

File: bin/email_notifications.js [hunk 1]
```diff
@@ -31,7 +31,7 @@ var deliveryQueue = new SQSReceiver(config.emailNotifications.region, [
 ])
 
 DB.connect(config[config.db.backend])
-  .done(
+  .then(
```
Comment (vladikoff) [inline bin/email_notifications.js]: @philbooth What was the reason for this?, seems like http://bluebirdjs.com/docs/api/done.html is still supported

Comment (vladikoff) [inline bin/email_notifications.js]: Ah sorry missed the 

> Unhandled rejections that reach done kill the process. To that end, I've replaced calls to done with then.

Comment (vladikoff) [inline bin/email_notifications.js]: Hopefully the process exists with status code >0

Comment (philbooth) [inline bin/email_notifications.js]: That's literally the first bullet point in the PR description! 😆

But, to repeat myself:

> * Unhandled rejections that reach `done` kill the process. To that end, I've replaced calls to `done` with `then`.

So it's a precautionary move really. Happy to change it back if you think I was wrong?

**Pull request merged**